docs/sql/sql.md

```diff
@@ -0,0 +1,498 @@
+> Its compatibility with future versions of SpacetimeDB is not guaranteed.
+
+```ebnf
+SHOW var
+```
+
+Returns the value of a system variable.
+
+## System Variables
+
+> NOTE: System variables are not part of the stable API.
+> Their compatibility with future versions of SpacetimeDB is not guaranteed.
+
+- `row_limit`
+
+    This system variable defines a cardinality threshold.
+    Queries are rejected if their estimated cardinality exceeds it.
+
+    Ex.
+    ```sql
+    -- Reject queries that read more than 10K rows
+    SET row_limit = 10000
+    ```
+
+## Data types
+
+SpacetimeDB data types are defined by [SATS],
+the Spacetime Algebraic Type System,
+and tuples are stored or encoded using [BSATN],
+the Binary Spacetime Algebraic Type Notation format.
+
+Spacetime SQL however does not support all of [SATS].
+In particular it has limited support for [products] and [sums].
+The language itself does not provide a way to construct them,
+nor does it provide scalar operations on top of them,
+the one exception being equality comparison.
+However the language does allow for reading them from tables,
+as well as returning them from queries.
+
+## Literals
+
+```ebnf
+literal = INTEGER | FLOAT | STRING | HEX | TRUE | FALSE ;
+```
+
+For each of the following [SATS] data types,
+we describe how to express their literal values in Spacetime SQL.
+
+### Bool
+
+In Spacetime SQL, the [boolean] Algebraic Type,
+representing the two truth values of boolean logic,
+is expressed using the canonical atoms `true` or `false`.
+
+### Integer
+
+The construction of [integer] values in Spacetime SQL.
+The concrete [SATS] type is inferred from the context.
+
+```ebnf
+INTEGER
+    = [ '+' | '-' ] NUM
+    | [ '+' | '-' ] NUM 'E' [ '+' | '-' ] NUM
+    ;
+
+NUM
+    = DIGIT { DIGIT }
+    ;
+
+DIGIT
+    = 0..9
+    ;
+```
+
+### Float
+
+The construction of [IEEE 754] [float] values in Spacetime SQL.
+The concrete [SATS] type is inferred from the context.
+
+```ebnf
+FLOAT
+    = [ '+' | '-' ] [ NUM ] '.' NUM
+    | [ '+' | '-' ] [ NUM ] '.' NUM 'E' [ '+' | '-' ] NUM
+    ;
+```
+
+### String
+
+The construction of variable length string values in Spacetime SQL,
+where `CHAR` is a `utf-8` encoded unicode character.
+
+```ebnf
+STRING
+    = "'" { "''" | CHAR } "'"
+    ;
+```
+
+### Hex
+
+Hexidecimal literals represent either [Identity] or [Address] types.
+The type is ultimately inferred from the context.
+
+```ebnf
+HEX
+    = 'X' "'" { HEXIT } "'"
+    | '0' 'x' { HEXIT }
+    ;
+
+HEXIT
+    = DIGIT | a..f | A..F
+    ;
+```
+
+## Identifiers
+
+```ebnf
+identifier
+    = LATIN { LATIN | DIGIT | '_' }
+    | '"' { '""' | CHAR } '"'
+    ;
+
+LATIN
+    = a..z | A..Z
+    ;
+```
+
+Identifiers are tokens that identify database objects like tables or columns.
+Spacetime SQL supports both quoted and unquoted identifiers.
+Quoted identifiers are case sensitive,
+whereas unquoted identifiers are case insensitive.
```

Comment:
This means that identifiers become lowercase like in PG.?

Reply:
Is there any issue with just making them all case sensitive? I assume that's not part of the SQL standard, but boy does it avoid some confusion.

Reply:
That's the only issue. We just wouldn't be SQL Standard compliant.

Reply:
Another option would be to nix unquoted identifiers altogether. This would make it so that everything is case sensitive, but we still maintain compliance with the standard. It's not a great UX though, since all identifiers, even column names, will need to be quoted.

Reply:
Again it's your call.

Reply:
Let's stick with the standard. So how you have it written.